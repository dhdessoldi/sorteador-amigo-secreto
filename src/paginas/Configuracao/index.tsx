import Formulario from "../../components/Formulario"
import ListaParticipantes from "../../components/ListaParticipantes"
import Rodape from "../../components/Rodape"
import styles from './Configuracao.module.scss'

const Configuracao = () => {
  return (
    <main className={styles.container}>
      <Formulario />
      <ListaParticipantes />
      <Rodape />
    </main>
  )
}

export default Configuracao