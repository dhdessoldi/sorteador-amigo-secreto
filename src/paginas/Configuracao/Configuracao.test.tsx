import { render } from "@testing-library/react";
import React from "react";
import { RecoilRoot } from "recoil";
import Configuracao from ".";

const mockNavegacao = jest.fn()

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavegacao
  }
})

describe('A página de configuração', () => {
  it('deve ser renderizada corretamente', () => {
    const { container } = render(<RecoilRoot><Configuracao /></RecoilRoot>)

    expect(container).toMatchSnapshot()

  })
})