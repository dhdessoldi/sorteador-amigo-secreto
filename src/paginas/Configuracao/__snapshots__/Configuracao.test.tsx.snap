// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A página de configuração deve ser renderizada corretamente 1`] = `
<div>
  <main
    class="container"
  >
    <form
      class="formulario"
    >
      <h1
        class="formulario__titulo"
      >
        Vamos começar!
      </h1>
      <div
        class="formulario__container"
      >
        <input
          class="formulario__campo"
          placeholder="Insira os nomes dos participantes"
          type="text"
          value=""
        />
        <button
          class="formulario__botao"
          disabled=""
        >
          Adicionar
        </button>
      </div>
      
    </form>
    <ul
      class="lista"
    />
    <footer
      class="footer"
    >
      <button
        class="footer__botao"
        disabled=""
      >
        Iniciar brincadeira
      </button>
      <div
        class="footer__imagem"
        role="img"
      />
    </footer>
  </main>
</div>
`;
